rtuclient.go

```diff
@@ -284,19 +286,24 @@ func (mb *rtuSerialTransporter) Send(aduRequest []byte) (aduResponse []byte, err
 	return
 }
 
+// charDuration returns time needed for a character to be transmitted.
+func (mb *rtuSerialTransporter) charDuration(chars float64) time.Duration {
+	return time.Duration(chars*float64(mb.DataBits+3)/float64(mb.BaudRate)) * time.Second
+}
+
 // calculateDelay roughly calculates time needed for the next frame.
 // See MODBUS over Serial Line - Specification and Implementation Guide (page 13).
 func (mb *rtuSerialTransporter) calculateDelay(chars int) time.Duration {
-	var characterDelay, frameDelay int // us
+	var characterDelay, frameDelay time.Duration
 
 	if mb.BaudRate <= 0 || mb.BaudRate > 19200 {
-		characterDelay = 750
-		frameDelay = 1750
+		characterDelay = mb.charDuration(1) + time.Duration(750)*time.Microsecond
```

Comment:
Minimalinvasiv:

    characterDelay = 10000000 / mb.BaudRate + 750

Reply:
Das wäre nur richtig wenn ein Char immer aus 10 Bits bestünde.

Die universelle Formel lautet aber
1 Startbit + x Datenbits + 0 oder 1 Paritybit + 1 oder 2 Stopbit

Modbus RTU schreibt aber quasi fest 11 vor, denn wenn man kein Parity-Bit benutzt wird sind streng nach Modbus RTU-Spec zwei Stopbits notwendig. Und was anderes als 8 Datenbits macht bei RTU wenig Sinn.

Das ist hier einer der Denkfehler in der Lib. Die magische Konstante ist hier einfach falsch - auch wenn es schön aussieht.

Der Standardmodus für RTU ist 8E1 mit folglich 11 Bits. Es ist aber in der Realität auch 8N1 anzutreffen obwohl es eigentlich 8N2 sein müsste...